octopusdeploy/schema_runbook_retention_period.go

```diff
@@ -0,0 +1,49 @@
+package octopusdeploy
+
+import (
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/runbooks"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func expandRunbookRetentionPolicy(flattenedRetentionPeriod interface{}) *runbooks.RunbookRetentionPeriod {
+	if flattenedRetentionPeriod == nil {
+		return nil
+	}
+
+	retentionPeriodProperties := flattenedRetentionPeriod.([]interface{})
+	if len(retentionPeriodProperties) == 1 {
+		retentionPeriodMap := retentionPeriodProperties[0].(map[string]interface{})
+		return &runbooks.RunbookRetentionPeriod{
+			QuantityToKeep:    int32(retentionPeriodMap["quantity_to_keep"].(int)),
+			ShouldKeepForever: retentionPeriodMap["should_keep_forever"].(bool),
+		}
+	}
+
+	return nil
+}
+
+func flattenRunbookRetentionPeriod(r *runbooks.RunbookRetentionPeriod) []interface{} {
+	retentionPeriod := make(map[string]interface{})
+	retentionPeriod["quantity_to_keep"] = int(r.QuantityToKeep)
+	retentionPeriod["should_keep_forever"] = r.ShouldKeepForever
+	return []interface{}{retentionPeriod}
+}
+
+func getRunbookRetentionPeriodSchema() map[string]*schema.Schema {
```

Comment:
Something is not right here, because it should be either the user specifies `quantity_to_keep` or `should_keep_forever`, but at the moment they can specify both!

Reply:
You can avoid this by using `conflictswith`

Reply:
:+1: This is a good catch. I'll make that change.